### ISSUE TITLE
linux docker functions deployed to azure 404 for http routes, fine in dev

### ISSUE THREAD

**Issue by vigouredelaruse** [Azure/azure-functions-docker#433]
i have a stock docker file generated against a stock azure durable functions orchestration that works sans docker in azure and locally, but fails http route discovery on docker deploy to azure while working docker local.

~~perhaps relevant- the below docker is deployed via azure devops ci/cd with a gratuitous double deploy where the container is pushed to a registry webhooked to the function, and then the azure deploy container step is applied.~~ 

FROM mcr.microsoft.com/azure-functions/dotnet:3.0-appservice AS base
WORKDIR /home/site/wwwroot
EXPOSE 80

etc.....
RUN dotnet publish "fxthatdon'tworkinazure.csproj" -c Release -o /app/publish

FROM base AS final
WORKDIR /home/site/wwwroot

COPY --from=publish /app/publish .

ENV AzureWebJobsScriptRoot=/home/site/wwwroot \
    AzureFunctionsJobHost__Logging__Console__IsEnabled=true

the above stock visual studio generated docker file depends on infrastructure serving from /home/site/wwwroot on http port 80

sadly deploying this to azure (b2 linux app svc plan east us) then navigating to the http routes deployed throws 404s that don't occur when not deployed to azure.

this is the relevant func azure functionapp logstream directly after a pull earlier today
2021-04-19T20:24:33.963407186Z info: Host.Startup[314]
2021-04-19T20:24:33.963455986Z       Loading functions metadata
2021-04-19T20:24:34.025849290Z info: Host.Startup[315]
2021-04-19T20:24:34.025907791Z       13 functions loaded
2021-04-19T20:24:34.256292960Z info: Host.Startup[0]
2021-04-19T20:24:34.256322960Z       Generating 13 job function(s)
2021-04-19T20:24:34.413041580Z info: Host.Startup[0]
2021-04-19T20:24:34.413067080Z       Found the following functions:

then as this is a durable orchestration the durable task things happen and expose secrets i won't show here but everything happens normally
2021-04-19T20:24:34.453352699Z
2021-04-19T20:24:34.480666751Z info: Host.Startup[412]
2021-04-19T20:24:34.480705751Z       Host initialized (525ms)
2021-04-19T20:24:34.511333746Z info: Host.Triggers.DurableTask[0]
2021-04-19T20:24:34.511586749Z       Starting task hub worker. InstanceId: . Function: . 
etc
021-04-19T20:24:35.463491315Z info: DurableTask.Core[11]
2021-04-19T20:24:35.463550116Z       Durable task hub worker is starting
2021-04-19T20:24:35.523951094Z info: DurableTask.AzureStorage[126]
2021-04-19T20:24:35.701217979Z info: DurableTask.Core[11]
2021-04-19T20:24:35.701271279Z       Durable task hub worker started successfully after 230ms
2021-04-19T20:24:35.711184007Z info: Host.Triggers.DurableTask[0]
2021-04-19T20:24:35.711524512Z       Task hub worker started. Latency: 00:00:01.1927412. I

note the following 'correct' docker deployment for the docker file deployed
2021-04-19T20:24:35.732273779Z info: Host.Startup[413]
2021-04-19T20:24:35.732326780Z       Host started (1797ms)
2021-04-19T20:24:35.732605983Z info: Host.Startup[0]
2021-04-19T20:24:35.732614583Z       Job host started
2021-04-19T20:24:35.868852339Z Hosting environment: Production
2021-04-19T20:24:35.868884439Z Content root path: /home/site/wwwroot
2021-04-19T20:24:35.868889839Z Now listening on: http://[::]:80
2021-04-19T20:24:35.868894039Z Application started. Press Ctrl+C to shut down.

but then after a pull
2021-04-19T20:24:05.057Z INFO  - Pull Image successful, Time taken: 0 Minutes and 7 Seconds
2021-04-19T20:24:05.077Z INFO  - Starting container for site
2021-04-19T20:24:05.078Z INFO  - docker run -d -p 7513:80 --name thecontainer

and shortly after the 'correct' settings are undone

2021-04-19T20:24:17.540Z INFO  - Starting container for site
2021-04-19T20:24:17.547Z INFO  - docker run -d -p 5375:8081 --name thecontainer

and now it's broken
nce initialized successfully and is ready to serve requests.

2021-04-19T20:23:32.791957542Z Hosting environment: Production
2021-04-19T20:23:32.792000142Z Content root path: /app
2021-04-19T20:23:32.792004342Z Now listening on: http://[::]:8081
2021-04-19T20:23:32.792007343Z Application started. Press Ctrl+C to shut down.

2021-04-19T20:29:06.145879901Z info: Host.Controllers.Host[0]
2021-04-19T20:29:06.145913102Z       Host Status: {
2021-04-19T20:29:06.145918002Z         "id": "it's running on the wrong port from the wrong siteroot",
2021-04-19T20:29:06.145922102Z         "state": "Running",

i'm blocked and  stumped. please advise

the environment is configured as per
![image](https://user-images.githubusercontent.com/31052867/115301293-fcfdef80-a12e-11eb-9016-634d76ea7ca3.png)

**Comment by balag0** [comment]
One possibility is WEBSITES_ENABLE_APP_SERVICE_STORAGE might be set to true on the app. In that case try removing that app setting or set it to false. If this is not the case, please share the appname.

You can use https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately if required,

**Comment by vigouredelaruse** [comment]
app-service-storage  is false... i'd appreciate you taking a look. is this enough info? i can't trigger the http functions (404) but i can trigger activities

2021-04-20T03:30:46.502717209Z         "state": "Running",
2021-04-20T03:30:46.502721309Z         "version": "3.0.15417.0",
2021-04-20T03:30:46.502725509Z         "versionDetails": "3.0.15417 Commit hash: a7ee1f99e07ce6efec991acd83b8e00306a9d693",
2021-04-20T03:30:46.502763110Z         "platformVersion": "93.0.7.75",
2021-04-20T03:30:46.502768210Z         "instanceId": "bcecd7525f4b7bd3dab05ad4f985c3361737e7748bf74832e0a2be875b273e64",
2021-04-20T03:30:46.502772310Z         "computerName": "RD0003FF50A269",
2021-04-20T03:30:46.502788010Z         "processUptime": 485672
2021-04-20T03:30:46.502791310Z       }
2021-04-20T03:30:46.734627404Z [40m[32minfo[39m[22m[49m: Host.Startup[314]
2021-04-20T03:30:46.734683305Z       Loading functions metadata
2021-04-20T03:30:46.742651911Z [40m[32minfo[39m[22m[49m: Host.Startup[315]
2021-04-20T03:30:46.742721712Z       13 functions loaded
2021-04-20T03:30:50.025783922Z [40m[32minfo[39m[22m[49m: Function.GetAllV2StorageAccounts[1]
2021-04-20T03:30:50.025816122Z       Executing 'GetAllV2StorageAccounts' (Reason='This function was programmatically called via the host APIs.', Id=a3a11ec4-1134-41ad-9e2d-42d327d63d16)
2021-04-20T03:30:50.340618323Z [41m[30mfail[39m[22m[49m: Function.GetAllV2StorageAccounts[3]
2021-04-20T03:30:50.340649123Z       Executed 'GetAllV2StorageAccounts' (Failed, Id=a3a11ec4-1134-41ad-9e2d-42d327d63d16, Duration=3153ms)

**Comment by balag0** [comment]
Looking at the exception from the above failing request, does the below help? Not an expert on Durable functions so not sure if this is something you can control.

GetAllV2StorageAccounts ---> System.InvalidOperationException : Exception binding parameter 'instanceId' ---> System.ArgumentException : The serialized input is expected to be an object array with one element.
    at Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableActivityContext.GetInputAsJson() at D:\a\r1\a\azure-functions-durable-extension\src\WebJobs.Extensions.DurableTask\ContextImplementations\DurableActivityContext.cs : 56
    at Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableActivityContext.GetInput(Type destinationType) at D:\a\r1\a\azure-functions-durable-extension\src\WebJobs.Extensions.DurableTask\ContextImplementations\DurableActivityContext.cs : 115

**Comment by vigouredelaruse** [comment]
nah the exception is due to a missing parameter i didn't provide for the call. this code works in all supported situations except in docker in azure. the failure mode is

- all deployed fx are mapped correctly - the set is orchestration, activity and http triggered fx
- only in docker in azure does the http triggered set fail to actually have valid routes, throwing 404
- the code works in all other supported deployments - that is in azure windows non-docker app svc, and azure linux app svc. docker works local but not in azure. there are no exceptions thrown on http trigger execution. only 404. the code is never run

note the deployed main proj file looks like below
![image](https://user-images.githubusercontent.com/31052867/115440559-df3b9380-a1dd-11eb-8140-7c763df1a7dd.png)


let me know if you need to see the output from a successful execution
thanks

**Comment by vigouredelaruse** [comment]
fixed - but on further reflection (later deployments) the 404 error returned.

thus i spun up a new function app docker instance, applied the relevant appsettings from the failing app (using the fresh deployment appsettings as a base) and the container that failed previously worked flawlessly on the new appservice instance (within the same appservice plan as the failing app)

so this seems actually related to a specific functionapp instance that has become 'poisoned' in some way - 

the bottom line is that the fix was to recreate the function app resource in azure
